File: core/src/main/java/javabot/commands/Param.java
package javabot.commands;

import static java.lang.annotation.ElementType.FIELD;
import java.lang.annotation.Retention;
import static java.lang.annotation.RetentionPolicy.RUNTIME;
import java.lang.annotation.Target;

@Retention(RUNTIME)
@Target({FIELD})
public @interface Param {
    String name() default "";

    boolean required() default true;

    boolean primary() default false;

    String defaultValue() default "";
}

File: src/main/kotlin/javabot/commands/Param.kt
package javabot.commands

import java.lang.annotation.ElementType.FIELD
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy.RUNTIME

Retention(RUNTIME)
@Target(AnnotationTarget.FIELD)
annotation public class Param(public val name: String = "", public val required: Boolean = true, public val primary: Boolean = false, public val defaultValue:

String = "")
